src/api/compose-types.js

```diff
@@ -0,0 +1,10 @@
+/**
+ * function that returns a function to apply the wrapper to a component while adding the proper types to your component via typescript
+ * @param {Function|Object} component - component default export
+ * @returns {Function} returns a function that takes the component to apply the wrappers to
+ */
+/* istanbul ignore next */
+export const composeTypes =
```

Comment:
Please reduce the scope of this PR to one single feature. I'd prefer to discuss this change into a separate PR thanks